executor/src/witgen/processor.rs

```diff
@@ -70,12 +73,36 @@ impl<'a, T: FieldElement> Processor<'a, T> {
         self.data[index] = self.row_factory.fresh_row();
     }
 
+    /// Figures out unknown values, using the default sequence iterator.
+    /// Since the default sequence iterator looks at the row before and after
+    /// the current block, we assume that these lines are already part of [Self::data]
```

Comment:
I think we should add assertions for that.

Reply:
Hmm, not sure where to add which assert.

I think the solution would be to change the semantics of "block size" of the sequence iterator to be consistent with that of the `Processor`: "block size" (perhaps renamed to `num_rows`) is just the number of rows we operate on. The block machine operates on `block_size + 2` rows (where here `block_size` is the period of the selector), because it also processes the row before and after each block. But that is a detail that shouldn't leak to the sequence iterator.

Reply:
I mean we could at least assert that `self.data.len() > 2`. Will add this.